### DISCUSSION TITLE
How does the Prisma data proxy pool connections properly if Prisma ORM uses prepared statements?

### DISCUSSION THREAD

**Discussion by eugene-kim** [prisma/prisma#15374] [Q&A] [answered]
Prisma has stated that AWS RDS Proxy isn't of much use since sessions are pinned, so I'd love to get more info on how the team solved this issue.

**Comment by janpio** [comment, marked as answer]
The problem is not the use of prepared statements itself, but that AWS RDS Proxy pins a connection as soon as a prepared statement was executed on it and then stops actually acting as a connection pooler. Data Proxy does not do that.

**Reply by eugene-kim** [reply]
Could you elaborate on how that works? Is Prisma Data Proxy running something like pgBouncer under the hood? There isn't much info online on what circumstances are ideal for Prisma data proxy so I'm unable to consider it as a potential solution.

**Reply by nurul3101** [reply]
Thank you for raising this, the feedback has been provided to the docs team to include information regarding how data proxy works.
Also for PDP-related queries you can also use the Zendesk channel by clicking on Support after logging in [Prisma Data Platform](https://cloud.prisma.io/)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
